numpydoc/docscrape_sphinx.py

```diff
@@ -219,6 +219,24 @@ def _obj(self):
             return self._f
         return None
 
+    def _get_class_that_defined_method(self, method):
+        try:
+            method_name = method.__name__
+            if method.__self__:
+                classes = [method.__self__.__class__]
+            else:
+                #unbound method
+                classes = [method.im_class]
+            while classes:
+                c = classes.pop()
+                if method_name in c.__dict__:
+                    return '~' + c.__module__ + '.' + c.__name__ + '.'
+                else:
+                    classes = list(c.__bases__) + classes
+            return ''
+        except:
+            return ''
```

Comment:
I think we should also simplify the docs to not include inherited attributes, i.e., set the option to `False` in our `conf.py` and remove this function.

Reply:
Not at all sure about this, i.e. `space.field` not showing would confuse quite some people. Many methods would basically have no methods (see `DiscreteLp`?)

My main suggestion would be that we drop the "one file per method" style, and switch to "one file per class". Easier to browse, and far less spam of pages.

Reply:
Also true, maybe just flattening the file hierarchy would already do a good job.

Reply:
I'd personally strongly prefer that approach